app/pages/api/buy.ts

```diff
@@ -0,0 +1,51 @@
+import { NextApiResponse, NextApiRequest } from "next";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== "POST") {
+    return res.status(404).json({ status: "Not found" });
+  }
+
+  if (!req.body) {
+    return res.status(500).json({ error: "Internal server error" });
+  }
+
+  const reqBody = JSON.parse(req.body);
+
+  if (!reqBody.address) {
+    return res.status(500).json({ error: "Internal server error" });
+  }
+
+  const mobilumUrl = new URL("https://api.mobilum.com/widget/GetWidget");
+  mobilumUrl.searchParams.set("walletAddress", reqBody.address);
+
+  const body = {};
+
+  const widgetResponse = await fetch(mobilumUrl.href, {
+    method: "POST",
+    headers: {
+      ApiKey: process.env.MOBILUM_API_KEY as string,
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(body),
+  });
+
+  if (!widgetResponse) {
+    return res.status(500).json({ error: "Internal server error" });
```

Comment:
if there is a network error, like Mobilum is down or our API key expired, `fetch` throws so it won't get to this point.
but if they give us a 500 or 404 or something instead, fetch does not throw

so we need to check `res.ok` and handle those cases

I also recommend using a "type guard" with `is` syntax to verify and apply the correct type to the response body

```ts
const isValidMobilumResponse = (body: unknown): body is MobilumResponse => {
  if (body?.result?.widgetBase64Html && body?.result?.widgetBase64ScriptUrl) return true;
  return false
}
```